Contributing.md

```diff
@@ -0,0 +1,177 @@
+* [Help wanted issues](https://github.com/novelrt/NovelRT/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) - issues which should be a bit more involved than `good first issues`.
+
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
+
+#### Local development
+
+For instructions on how to get started with local development of NovelRT, see the following sections in [the engine's README](https://github.com/novelrt/NovelRT#getting-started-with-novelrt).
+
+### Pull Requests
+
+The process described here has several goals:
+
+- Maintain NovelRT's quality
+- Fix problems that are important to users
+- Engage the community in working toward the best possible NovelRT
+- Enable a sustainable system for NovelRT's maintainers to review contributions
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+1. Ensure that the title describes the overall enhancement/fix that is being provided.
+2. The description should summarise the changes being made, and should call out what area(s) of the engine is being affected.
+3. Follow the styleguide as provided by our [`.clang-format` file](https://github.com/novelrt/NovelRT/blob/main/.clang-format). (Our CI will assist you if you are unable to proactively do so, and will provide a git patch to fix most formatting issues, however this is not perfect and should not be guaranteed.)
+4. The pull request that is opened should be [linked to a corresponding issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
+5. Once a pull request is opened, if you have not done so already you will be asked to sign our Contributor License Agreement (CLA).
+5. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing <details><summary>What if the status checks are failing?</summary>If a status check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our status check suite.</details>
+
+While the prerequisites above must be satisfied prior to having your pull request reviewed, the reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted. Also, please note that under the circumstances listed below, a pull request _may_ be rejected with/without further recourse by the NovelRT team for the following:
+
+* Creating a pull request without an open issue (**_this is limited to contributors only - outside collaborators and maintainers are permitted within reason_**).
```

Comment:
I'd probably disagree with this for the sole purpose of documentation - IMO creating a whole issue just to say "i think `teh` is a typo of `the` in this docs page" is a bit too much. Maybe we should clarify that we're specifically talking about code/functionality changes here?

Reply:
That works for me (calling out code/functionality changes w/o a PR)..

@RubyNova your take? You have final say, but I'm in @FiniteReality's boat here

Reply:
My only real concerns are if this causes conflicts in headers and creation of misaligned expectations. Logistically speaking it would be easier on us if someone who we knew was already editing a header file made minor documentation changes to make it easier on the merge conflicts. In addition to this, I'm just mildly concerned someone will be like "x got merged without an issue why not y" and then you're on a slippery slope.

How do we address these?

Reply:
TL;DR - Overall, I'd propose a matching Pull Request Template and Issue template (not in this PR but in a new one). Would address the issues, and would only require communication in Discord/GH prior to a PR being accepted.


Long version - Well, this depends on how we want to handle this holistically imo.

To be honest, after thinking it further, typos and such are still issues, therefore unless someone trusted is working on it, the person pushing for a fix should _probably_ just open the matching issue for it. On the contrary this is _going_ to inflate our issue count if we ever run into an issue where we're finding lots of typos, etc, so we'd run into the exact scenario that @FiniteReality mentioned. 
But at least if someone trusted is working on it, there's typically more communication in at least Discord, therefore there's less chance of collision / merge conflicts @RubyNova.

As far as this goes:
> "x got merged without an issue why not y"
We'd really just need to be diligent on ensuring contributors know what was being affected (documentation only) and _whom_ was affecting it (if going the route of letting documentation-style quick fixes go without a matching issue). 

With that said, we could introduce templates for PRs and Issues, and only require that code/functionality changes have a matching issue _as long as the template is followed and communication has occurred somewhere publicly visible_ with [insert someone with merge rights here]. This gives us an easier way to handle - anyone not following the template can be closed until they're following the format (once they're introduced - and we can have GH help notify contributors of this).

Reply:
Opened #426 to track this work - won't resolve until I reword some of this as per Monica's original suggestions.

Reply:
Resolved (hopefully) in latest commit